### PR TITLE
Update Client tool to support custom TEE pubkey and KBS certificate

### PR DESCRIPTION
**Pull request by jialez0**
1. Added optional parameters for importing custom TEE public key.
2. Added optional parameters for provide custom KBS Root certificate.
3. Provided SDK format for Client Tool.

These enhancements will make the Client tool easier to use

### CODE REVIEW

File: tools/client/Cargo.toml [hunk 1]
```diff
@@ -13,7 +20,7 @@ base64.workspace = true
 clap = { version = "4.0.29", features = ["derive"] }
 env_logger.workspace = true
 jwt-simple = "0.11.4"
-kbs_protocol = { git = "https://github.com/confidential-containers/guest-components", rev = "43f6832" }
+kbs_protocol = { git = "https://github.com/jialez0/guest-components", branch = "https" }
```
Comment (jialez0) [inline tools/client/Cargo.toml:23]: Here, we are waiting https://github.com/confidential-containers/guest-components/pull/273, for now temporarily pointing to the branch of downstream.

**Comment by sameo** [comment]
The PR looks good to me. Do we want to wait for https://github.com/confidential-containers/guest-components/pull/273 to be merged or you'd like this to go in sooner?

File: tools/client/README.md [hunk 2]
```diff
@@ -19,3 +17,11 @@ If you want to use Sample TEE attester in CC-KBC, set the following environment
...
 export AA_SAMPLE_ATTESTER_TEST=yes
 ```
+
+## Client SDK
+
+Add the following dependencies to the `Cargo.toml` of the Rust project to import KBS Client SDK:
+
+```
+kbs_client = { git = "https://github.com/confidential-containers/kbs" }
+```
```
Comment (sameo) [inline tools/client/README.md:27]: I'd prefer to have that one as a separate PR.

**Review by sameo — APPROVED** [review]

**Comment by jialez0** [comment]
> The PR looks good to me. Do we want to wait for [confidential-containers/guest-components#273](https://github.com/confidential-containers/guest-components/pull/273) to be merged or you'd like this to go in sooner?

@sameo We can wait for https://github.com/confidential-containers/guest-components/pull/273 to be merged. This is definitely after the release of v0.7.0, we don't plan to add this PR's feature in v0.7.0.

File: tools/client/README.md (hunk 2 above)
Comment (jialez0) [reply tools/client/README.md:27]: Okay, I'll separate it.

**Comment by sameo** [comment]
> > The PR looks good to me. Do we want to wait for [confidential-containers/guest-components#273](https://github.com/confidential-containers/guest-components/pull/273) to be merged or you'd like this to go in sooner?
> 
> @sameo We can wait for [confidential-containers/guest-components#273](https://github.com/confidential-containers/guest-components/pull/273) to be merged. This is definitely after the release of v0.7.0, we don't plan to add this PR's feature in v0.7.0.

Ok, so maybe we should tag this PR as a draft, or WIP, so that we don't merge it before the guest components PR is merged as well?

**Pull request merged**